pkg/agent/proxy/proxier.go

```diff
@@ -744,6 +755,13 @@ func (p *proxier) installServices() {
 				pSvcInfo.ExternalPolicyLocal() != svcInfo.ExternalPolicyLocal() ||
 				pSvcInfo.InternalPolicyLocal() != svcInfo.InternalPolicyLocal()
 			if p.cleanupStaleUDPSvcConntrack && needClearConntrackEntries(pSvcInfo.OFProtocol) {
+				// We clean the UDP conntrack entries when Service IPs or ports are changed. However, we DO NOT clean
+				// the UDP conntrack entries related to remote Endpoints that are still referenced by the Service but
+				// are no longer selectable Endpoints for the corresponding Service IPs (for externalTrafficPolicy,
+				// these IPs are loadBalancerIPs, externalIPs and NodeIPs; for internalTrafficPolicy, these IPs
+				// clusterIPs) when externalTrafficPolicy or internalTrafficPolicy is changed from Cluster to Local.
+				// Consequently, the connections, which are supposed to select local Endpoints, will continue send
+				// packets to remote Endpoints due to the existing UDP conntrack entries until they timeout.
```

Comment:
After reading this, I feel the current implementation is not strictly correct as it would clean all connections even when a particular kind of IP changes? For example, if it gets an LoadBalancerIP assigned , it shouldn't clean up the connections of ClusterIP?

Reply:
We only do the delta deletion. We won't clean up the connections of ClusterIP if it doesn't change. (the ClusterIP is an IP allocated from Service CIDR if I understand it correctly).

Reply:
Ah, thanks for pointing it out. Then we are good.